lib/checkfunctions.cpp

```diff
@@ -218,7 +218,11 @@ void CheckFunctions::checkIgnoredReturnValue()
             if ((!tok->function() || !Token::Match(tok->function()->retDef, "void %name%")) &&
                 (mSettings->library.isUseRetVal(tok) || (tok->function() && tok->function()->isAttributeNodiscard())) &&
                 !WRONG_DATA(!tok->next()->astOperand1(), tok)) {
-                ignoredReturnValueError(tok, tok->next()->astOperand1()->expressionString());
+                const auto retvalTy = mSettings->library.getUseRetvalErrorType(tok);
```

Comment:
If you replace `auto` with `Library::RetvalErrorType` then it will be easier to see what type is used imho. My IDE does not tell me which type auto has..